pkg/stub/handler.go

```diff
@@ -62,16 +62,49 @@ func (h *Handler) EnsureDefaultClusterIngress() error {
 		return err
 	}
 
-	err = sdk.Create(ci)
-	if err == nil || errors.IsAlreadyExists(err) {
-		if err == nil {
-			logrus.Infof("created default cluster ingress %s/%s", ci.Namespace, ci.Name)
+	changed, nci, err := checkClusterIngress(ci)
+	if err != nil {
+		return err
+	}
+	if changed {
+		err = sdk.Update(nci)
+		if err != nil {
+			return fmt.Errorf("updating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
 		}
-
-		return nil
+		logrus.Infof("updated default cluster ingress %s/%s", ci.Namespace, ci.Name)
+	} else if nci == nil {
+		err = sdk.Create(ci)
+		if err != nil {
+			return fmt.Errorf("creating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
+		}
+		logrus.Infof("created default cluster ingress %s/%s", ci.Namespace, ci.Name)
 	}
+	return nil
+}
 
-	return fmt.Errorf("creating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
+func checkClusterIngress(ci *ingressv1alpha1.ClusterIngress) (bool, *ingressv1alpha1.ClusterIngress, error) {
+	oldci := &ingressv1alpha1.ClusterIngress{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       ci.Kind,
+			APIVersion: ci.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ci.Name,
+			Namespace: ci.Namespace,
+		},
+	}
+	err := sdk.Get(oldci)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return false, nil, fmt.Errorf("failed to fetch existing default cluster ingress %s/%s, %v", ci.Namespace, ci.Name, err)
+		}
+		return false, nil, nil
+	}
+	if *oldci.Spec.IngressDomain != *ci.Spec.IngressDomain {
```

Comment:
If either of this is `nil` will the function panic?

Reply:
Let me fix this